### PR TITLE
Fix ButtonGroup get_pressed_button in toggle mode

### LINKED ISSUE #25057: ButtonGroup.get_pressed_button returns "null" or previous button in toggle mode
**Godot version:**
9c17688

**OS/device including version:**
Windows10

**Issue description:**
Creating a number of buttons and assigning them to the same button group. Connecting their "pressed" signal to "on_pressed" function (function just needs their index, so no need to make a function for each button). ButtonGroup.get_pressed_button only works if toggle_mode on button is false, contrary to the doc.

**Steps to reproduce:**
Open attached project
run, click buttons and check output
setting button.toggle_mode to "false" makes it work properly

**Minimal reproduction project:**
[button_group_bug.zip](https://github.com/godotengine/godot/files/2767664/button_group_bug.zip)

### PR DESCRIPTION
**Pull request by 99alfie**
resubmitting the pr after syncing with the base repo

*Bugsquad edit:* Fixes #25057.

### CODE REVIEW

File: scene/gui/base_button.cpp [hunk 1]
```diff
@@ -151,7 +153,6 @@ void BaseButton::_gui_input(Ref<InputEvent> p_event) {
 					emit_signal("toggled", status.pressed);
 				}
```
Comment (SoIAS) [inline scene/gui/base_button.cpp]: You should remove this empty line to comply with clang format

File: scene/gui/base_button.cpp [hunk 2]
```diff
@@ -230,7 +233,6 @@ void BaseButton::_gui_input(Ref<InputEvent> p_event) {
...
```
Comment (SoIAS) [inline scene/gui/base_button.cpp]: This one too.

**Comment by akien-mga** [comment]
Could you squash commits together? See [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html).

**Comment by 99alfie** [comment]
sorry I created a mess by syncing my repo with the base in the middle. I should remove that first.

**Pull request closed without merging**